.pipelines/UpdateVersions.ps1

```diff
@@ -0,0 +1,97 @@
+$nugetOutput = nuget list Microsoft.WindowsAppSDK `
+    -Source "https://microsoft.pkgs.visualstudio.com/ProjectReunion/_packaging/Project.Reunion.nuget.internal/nuget/v3/index.json" `
+    -AllVersions
+
+$filteredVersions = $nugetOutput | Where-Object { $_ -match "Microsoft.WindowsAppSDK 1\.6\." }
+# Write-Host "Filtered versions: $filteredVersions"
+$latestVersions = $filteredVersions | Sort-Object { [version]($_ -split ' ')[1] } -Descending | Select-Object -First 1
+Write-Host "Latest versions found: $latestVersions"
+
+# Extract the latest version number from the output
+$latestVersion = $latestVersions -split "`n" | `
+    Select-String -Pattern 'Microsoft.WindowsAppSDK\s*([0-9]+\.[0-9]+\.[0-9]+-*[a-zA-Z0-9]*)' | `
+    ForEach-Object { $_.Matches[0].Groups[1].Value } | `
+    Sort-Object -Descending | `
+    Select-Object -First 1
+
+if ($latestVersion) {
+    $WinAppSDKVersion = $latestVersion
+    Write-Host "Extracted version: $WinAppSDKVersion"
+    Write-Host "##vso[task.setvariable variable=WinAppSDKVersion]$WinAppSDKVersion"
+} else {
+    Write-Host "Failed to extract version number from nuget list output"
+    exit 1
+}
+
+# Update packages.config files
+Get-ChildItem -Recurse packages.config | ForEach-Object {
+    $content = Get-Content $_.FullName -Raw
+    if ($content -match 'package id="Microsoft.WindowsAppSDK"') {
+        $newVersionString = 'package id="Microsoft.WindowsAppSDK" version="' + $WinAppSDKVersion + '"'
+        $oldVersionString = 'package id="Microsoft.WindowsAppSDK" version="[-.0-9a-zA-Z]*"'
+        $content = $content -replace $oldVersionString, $newVersionString
+        Set-Content -Path $_.FullName -Value $content
+        Write-Host "Modified " $_.FullName 
+    }
+}
+
+# Update Directory.Packages.props file
+$propsFile = "Directory.Packages.props"
+if (Test-Path $propsFile) {
+    $content = Get-Content $propsFile -Raw
+    if ($content -match '<PackageVersion Include="Microsoft.WindowsAppSDK"') {
+        $newVersionString = '<PackageVersion Include="Microsoft.WindowsAppSDK" Version="' + $WinAppSDKVersion + '" />'
+        $oldVersionString = '<PackageVersion Include="Microsoft.WindowsAppSDK" Version="[-.0-9a-zA-Z]*" />'
+        $content = $content -replace $oldVersionString, $newVersionString
+        Set-Content -Path $propsFile -Value $content
+        Write-Host "Modified " $propsFile
+    }
+}
+
+# Update .vcxproj files
+Get-ChildItem -Recurse *.vcxproj | ForEach-Object {
+    $content = Get-Content $_.FullName -Raw
+    if ($content -match '\\Microsoft.WindowsAppSDK.') {
+        $newVersionString = '\Microsoft.WindowsAppSDK.' + $WinAppSDKVersion + '\'
+        $oldVersionString = '\\Microsoft.WindowsAppSDK.[-.0-9a-zA-Z]*\\'
+        $content = $content -replace $oldVersionString, $newVersionString
+        Set-Content -Path $_.FullName -Value $content
+        Write-Host "Modified " $_.FullName
+    }
+}
+
+# Update .csproj files
+Get-ChildItem -Recurse *.csproj | ForEach-Object {
```

Comment:
is this just being safe in the future?

Reply:
Just got the idea from the pipeline in WindowsAppSDK-Samples, it also did it in this way: https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/UpdateVersions.ps1